src/lib.rs

```diff
@@ -473,20 +476,55 @@ impl Account {
     ) -> Result<(Account, AccountCredentials), Error> {
         Self::create_inner(
             account,
+            Key::generate()?,
             external_account,
             Client::new(server_url, http).await?,
             server_url,
         )
         .await
     }
 
+    /// Load a new account by private key, with a default or custom HTTP client
+    ///
+    /// https://www.rfc-editor.org/rfc/rfc8555#section-7.3.1
+    ///
+    /// The returned [`AccountCredentials`] can be serialized and stored for later use.
+    /// Use [`Account::from_credentials()`] to restore the account from the credentials.
+    pub async fn load(
+        private_key_pkcs8_der: &[u8],
+        server_url: &str,
+        http: Option<Box<dyn HttpClient>>,
+    ) -> Result<(Account, AccountCredentials), Error> {
+        let client = match http {
+            Some(http) => Client::new(server_url, http).await?,
+            None => Client::new(server_url, Box::new(DefaultClient::try_new()?)).await?,
+        };
+
+        let key = Key::from_pkcs8_der(private_key_pkcs8_der)?;
```

Comment:
I think we should probably externalize this, instead make `Key::from_pkcs8_der()` public, and have this function take a `Key`.

Reply:
I first did it that way, but wasn't sure you were open to extend the area of public API.

Reply:
But to be honest, if we are making Key part of the public API, it would make more sense to add the Key as the param of the Account::create as well, so that the user has the ability to create an account with a chosen key instead of forcing to generate one for them.

And in that case for_existing_account would probably just be a convenience wrapper around that.